core/src/main/java/com/linecorp/armeria/server/ServiceNotFoundException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import com.linecorp.armeria.common.util.Exceptions;
+
+/**
+ * A {@link RuntimeException} raised when a {@link Service} failed to find a resource.
+ */
+public final class ServiceNotFoundException extends RuntimeException {
```

Comment:
`DynamicHttpService` already knows if the request path will match its mappings or not even before generating any response, so I think we can even implement this behavior without introducing this exception.

For example, we could improve the `OrElseService` to look up the mappings of the `DynamicHttpService` so that it delegates the request to the second service even before trying `DynamicHttpService.serve()`. WDYT?